### ISSUE TITLE
[Sema] allocatable statement issuing wrong error

### ISSUE THREAD

**Issue by clementval** [flang-compiler/f18#930]
The following code is normally correct according to Fortran standard. But f18 is reporting the following error: 

Error
```bash
dummy.f90:2:17: error: Array 'c' without ALLOCATABLE or POINTER attribute must have explicit shape
    integer :: a, c(:,:,:)
                  ^
dummy.f90:6:26: error: Shape specifications must not appear when allocatable object is scalar
    allocate(a(2), b(2,2), c(2,2,2))
                           ^
./install/bin/f18: semantic errors in dummy.f90
```

Fortran code
```fortran
subroutine sub()
  integer :: a, c(:,:,:)
  integer, allocatable :: b(:,:)
  allocatable a(:), c

  allocate(a(2), b(2,2), c(2,2,2))
  deallocate(a,b,c)
end subroutine
```
